FirebasePerformance/Sources/AppActivity/FPRAppActivityTracker.m

```diff
@@ -147,6 +160,36 @@ - (FIRTrace *)activeTrace {
   return self.backgroundSessionTrace;
 }
 
+- (void)startTrackingNetwork {
+  self.networkType = firebase_perf_v1_NetworkConnectionInfo_NetworkType_NONE;
+
+  __weak FPRAppActivityTracker *weakSelf = self;
+  if (@available(iOS 12, tvOS 12, *)) {
+    dispatch_queue_attr_t attrs = dispatch_queue_attr_make_with_qos_class(
+        DISPATCH_QUEUE_SERIAL, QOS_CLASS_UTILITY, DISPATCH_QUEUE_PRIORITY_DEFAULT);
+    weakSelf.monitorQueue =
+        dispatch_queue_create("com.google.firebase.perf.network.monitor", attrs);
+
+    weakSelf.monitor = nw_path_monitor_create();
+    nw_path_monitor_set_queue(weakSelf.monitor, weakSelf.monitorQueue);
+    nw_path_monitor_set_update_handler(weakSelf.monitor, ^(nw_path_t _Nonnull path) {
```

Comment:
I think use monitor here is fine: 
```
nw_path_monitor_set_update_handler(monitor, ^(nw_path_t _Nonnull path) {
 // within use weakSelf
}
```

Reply:
I don't monitor to be a local variable sicne that has the risk of being lost in the stack rather than being in the heap.